front/src/common/uploadFileModal.tsx

```diff
@@ -41,21 +46,18 @@ const UploadFileModal = ({ handleSubmit }: UploadFileModalProps) => {
         try {
           JSON.parse(await fileToValidate.text());
         } catch (e) {
+          console.error(e);
           return t('errorMessages.errorInvalidJSONFormat').toString();
         }
       } else if (
         fileToValidate.type === 'application/railml' ||
         fileToValidate.name.endsWith('.railml')
       ) {
         const fileContent = await fileToValidate.text();
-        const xmlError = parseXML(fileContent);
-        if (xmlError) {
-          return xmlError;
-        }
+        parseXML(fileContent);
```

Comment:
I kept it because it seems it can still throw an error if some conditions are met

Reply:
What I mean is that the original code worked fine, no?

With the new code, `validateFile()` no longer returns an error when it fails. There are cases where it returns `undefined` on XML parsing errors.

Reply:
Will probably wait for this `[PR](https://github.com/OpenRailAssociation/osrd/pull/10139)` to be merged because a lot of things are changing in these files

Reply:
The string is supposed to be displayed under the file input. See `isValid`.

I don't think an XML format error should be treated differently from an unsupported file type error, a JSON error or an empty file error. Either all should display a string under the input (status quo), either all should show an error bubble.